src/core/service.c

```diff
@@ -3188,9 +3188,9 @@ static int service_deserialize_item(Unit *u, const char *key, const char *value,
                         s->reload_result = f;
 
         } else if (streq(key, "control-pid")) {
-                pidref_done(&s->control_pid);
 
-                (void) deserialize_pidref(fds, value, &s->control_pid);
+                if (!pidref_is_set(&s->control_pid))
+                        (void) deserialize_pidref(fds, value, &s->control_pid);
```

Comment:
Please also do the same for `main-pid`.

Reply:
@yuwata `main-pid` is already handled by `service_set_main_pidref()`

Reply:
Ah, indeed. Sorry for the noise. LGTM.

Reply:
This is wrong... I'll submit a follow-up.

Reply:
Oh, really? My bad...